### ISSUE TITLE
`#` in branch name breaks nixci

### ISSUE THREAD

**Issue by srid** [srid/nixci#17]
```
❯ nix run nixpkgs#nixci https://github.com/nammayatri/nammayatri/pull/2844
🍏 github:nammayatri/nammayatri/backend/bug-fix/#916-Driver-Rating-Customer
Error: nix eval failed to run (exited: 1)
```

**Comment by srid** [comment]
stderr after 69a35d455ad4f607537c20f4cfcd88d6b7a998cd

```
🍏 github:nammayatri/nammayatri/backend/bug-fix/#916-Driver-Rating-Customer
Error: nix eval failed to run (exited: 1). raw_output = error: 'github:nammayatri/nammayatri/backend/bug-fix/#916-Driver-Rating-Customer#nixci' is not a valid URL
```

**Comment by srid** [comment]
`git+https://github.com/nammayatri/nammayatri?ref=backend/bug-fix/%23916-Driver-Rating-Customer` works, although it is rather verbose.
